WireDomain/Sources/WireDomain/Synchronization/OneOnOneResolver.swift

```diff
@@ -0,0 +1,306 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import CoreData
+import WireAPI
+import WireDataModel
+
+// sourcery: AutoMockable
+/// Resolves 1:1 conversations
+protocol OneOnOneResolverProtocol {
+    func invoke() async throws
+}
+
+struct OneOnOneResolver: OneOnOneResolverProtocol {
+
+    private enum OneOnOneResolverUseCaseError: Error {
```

Comment:
```suggestion
    private enum OneOnOneResolverError: Error {
```

Reply:
renamed to Error in https://github.com/wireapp/wire-ios/pull/1999/commits/25021b3841574291cb480d46d50180fec40f9654